File: hw/viteri/u2/HW15MongoShellAndCoding/HW15MongoShellAndCodeInPyton/HW15MongoShellAndCode.py
from pymongo import MongoClient
from pymongo.collection import Collection

def main():
    # Reemplaza esto con tu URI de conexión de MongoDB Atlas
    uri = "mongodb+srv://alexisviterigithub:ajviteri2@ajviteri2.y5pwei7.mongodb.net/"

    # Crear un cliente de MongoDB
    try:
        client = MongoClient(uri)
        
        # Seleccionar la base de datos
        database = client["oop"]
        
        # Seleccionar la colección
        collection: Collection = database["student"]

        # Leer los datos del usuario
        id = input("Id: ")
        name = input("Name: ")

        male_input = input("Male (True or False): ").strip().lower()
        male = male_input == 'true'

        sibling = int(input("Siblings: "))

        # Crear un documento
        student = {
            "id": id,
            "name": name,
            "male": male,
            "sibling": sibling
        }

        # Insertar el documento en la colección
        collection.insert_one(student)

        print("Document is updated")

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()

File: hw/segarra/u2/HW15MongoShell&Code/HW15JavaScriptMongoShellAndCode/ec/edu/espe/HW15Code/view/HW15MongoShellAndCode.js
//HW15MongoShellAndCode.js

const { MongoClient } = require('MongoDB');

async function main() {
    const uri = 'mongodb+srv://segarra:segarra@cluster0.b2q6ac3.mongodb.net/';
    const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });

    try {
        
        await client.connect();

        
        const database = client.db('oop');
        const collection = database.collection('students');

        const newStudent = {
            id: 'L00428992',
            name: 'Eduardo Segarra',
            male: true,
            siblings: 1
        };

        // Inserta el objeto en la colección
        const result = await collection.insertOne(newStudent);

        console.log(`Nuevo estudiante insertado con el id: ${result.insertedId}`);
    } catch (err) {
        console.error(err);
    } finally {
        // Cierra la conexión con MongoDB
        await client.close();
    }
}

main().catch(console.error);